pkg/connector/storage_account.go

```diff
@@ -118,7 +118,7 @@ func (usr *storageAccountBuilder) Grants(ctx context.Context, resource *v2.Resou
 		client, err := armauthorization.NewRoleAssignmentsClient(
 			storageResourceIDs.subscriptionID,
 			usr.conn.token,
-			nil,
+			usr.conn.client.ArmOptions(),
```

Comment:
Would you be able to add an explanation in the PR description for why this fixes the 404? (so we can keep track of what the bug fixes are without needing context) 

my understanding is that this allows us to override the http client and use the uhttp client, which will treat the 404s as just warnings, not errors, is my understanding correct?

Reply:
> my understanding is that this allows us to override the http client and use the uhttp client, which will treat the 404s as just warnings, not errors, is my understanding correct?

Right.

Also I will add more context in the PR description